### PR TITLE
feat(docs): add allowHierarchicalFolderAssignment config flag descrip…

### PR DESCRIPTION
**Pull request by tobiasweiss93**
…tion

Concerns CU-76u9kc

### CODE REVIEW

**Comment by meshcloud-gmbh** [comment]
Task linked: [CU-76u9kc Consider hierarchical folder structure in replication](https://app.clickup.com/t/76u9kc)

File: docs/meshstack.gcp.index.md [hunk 1]
```diff
@@ -206,12 +206,17 @@ let GcpPlatformCoreConfiguration =
         The resulting string must not exceed a total length of 30 characters. Only alphanumeric + hyphen are allowed.
         We recommend that configuration include at least 3 characters of the random parameter to reduce the chance of
         naming collisions.
+
+      allowHierarchicalFolderAssignment:
```
Comment (chewbeccy) [inline docs/meshstack.gcp.index.md:210]: @tobiasweiss93 can you give a little bit more insight in the impact of the replication behaviour.

Comment (tobiasweiss93) [reply docs/meshstack.gcp.index.md:210]: I've extended the description.

File: docs/meshstack.gcp.index.md [hunk 2]
```diff
@@ -206,12 +206,18 @@ let GcpPlatformCoreConfiguration =
...
+        Configuration flag to enable or disable hierarchical folder assignment in GCP. This means
+        projects can be assigned to sub folders of the defined resource manager folder in a landing zone. 
+        If this config flag is disabled the replicator force to create projects only directly under the resource manager folder.
```
Comment (chewbeccy) [inline docs/meshstack.gcp.index.md:213]: I thought the replicator would also move a project to the parent folder. Not only enforce this during project creation. Think about imported tenants via meshObject import.

Comment (tobiasweiss93) [reply docs/meshstack.gcp.index.md:213]: I've talked about this with Stefan. With this flag we only verify that a project is attached to a sub folder of the defined resource. If this is not the case then we call the update project function. After this step the project will be moved to the right place depend on the defined gcp function.

**Review by chewbeccy — APPROVED** [review]

**Pull request merged**